feat(route): add 哈尔滨工业大学研究生院通知公告

hlmu opened the pull request:
<!-- 
如有疑问，请参考 https://github.com/DIYgod/RSSHub/discussions/8002
Reference: https://github.com/DIYgod/RSSHub/discussions/8002
-->

## 完整路由地址 / Example for the proposed route(s)

<!--
请在 `routes` 区域填写以 / 开头的完整路由地址，否则你的 PR 将会被无条件关闭。
如果路由包含在文档中列出可以完全穷举的参数（例如分类），请依次全部列出。

Please include route starts with /, with all required and optional parameters. Fail to comply will result in your pull request being closed automatically.
```route
/some/route
/some/other/route
/dont/use/this/or/modify/it
/use/the/fenced/code/block/below
```
如果你的 PR 与路由无关, 请在 `routes` 区域 填写 `NOROUTE`，而不是直接删除 `routes` 区域。否则你的 PR 将会被无条件关闭。
If your changes are not related to route, please fill in `routes` with `NOROUTE`. Fail to comply will result in your PR being closed.
-->

```routes
/hit/hitgs
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [x] 新的路由 New Route
  - [ ] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [x] 全文获取 fulltext
  - [x] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note

hlmu committed:
添加"哈尔滨工业大学研究生院通知公告"路由

hlmu committed:
Merge branch 'master' of github.com:hlmu/RSSHub

TonyRL reviewed (changes requested):
New route should follow [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)

hlmu commented:
> New route should follow [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)

哈尔滨工业大学通知公告已有采用v1的路由
`hit/today`
`hit/jwc`
因此新的`hit/hitgs`同样采用v1路由以方便后面同学维护，请问这样可以吗？

TonyRL commented:
No exceptions
https://github.com/DIYgod/RSSHub/pulls?q=is%3Apr+is%3Aclosed+label%3A%22Route%3A+v1%22+label%3A%22Route%3A+v2%22

hlmu committed:
修改路由为v2 script

TonyRL commented on lib/v2/hit/hitgs.js:
```diff
@@ -0,0 +1,64 @@
+const cheerio = require('cheerio');
+
+module.exports = async (ctx) => {
+    const host = 'http://hitgs.hit.edu.cn';
```
```suggestion
    const host = 'https://hitgs.hit.edu.cn';
```
HTTPS is available

TonyRL commented on lib/v2/hit/hitgs.js:
```diff
@@ -0,0 +1,64 @@
+module.exports = async (ctx) => {
+    const host = 'http://hitgs.hit.edu.cn';
+
+    const response = await got({
+        method: 'get',
+        url: host + '/tzgg/list.htm',
+        headers: {
+            Referer: host,
+        },
+    });
```
got request can be simplified to
```suggestion
    const response = await got(host + '/tzgg/list.htm', {
        headers: {
            Referer: host,
        },
    });
```

TonyRL commented on lib/v2/hit/hitgs.js:
```diff
@@ -0,0 +1,64 @@
+    const $ = cheerio.load(response.data);
+    const list = $('.news_list li')
+        .map((i, e) => ({
+            path: $('span a', e).attr('href'),
```
You can prepare the `link` property here
```suggestion
            link: host + $('span a', e).attr('href'),
```

TonyRL commented on lib/v2/hit/hitgs.js:
```diff
@@ -0,0 +1,64 @@
+    const list = $('.news_list li')
+        .map((i, e) => ({
+            path: $('span a', e).attr('href'),
+            pubDate: new Date($('span:nth-child(4)', e).text()).toUTCString(),
```
Please use `parseDate`
docs: https://docs.rsshub.app/joinus/pub-date.html
```suggestion
            pubDate: parseDate($('span:nth-child(4)', e).text()),
```

TonyRL commented on lib/v2/hit/hitgs.js:
```diff
@@ -0,0 +1,64 @@
+        }))
+        .get();
+
+    const out = await Promise.all(
+        list.map(async (item) => {
+            const { path, pubDate, title, author } = item;
+            const link = host + path;
+
+            const cache = await ctx.cache.get(link);
+            if (cache) {
+                return Promise.resolve(JSON.parse(cache));
+            }
+
+            const response = await got({
+                method: 'get',
+                url: link,
+                headers: {
+                    Referer: host,
+                },
+            });
+
+            const $ = cheerio.load(response.data);
+            const description = $('.wp_articlecontent').html();
+            const single = {
+                pubDate,
+                author,
+                link,
+                title,
+                description,
+            };
+
+            ctx.cache.set(link, JSON.stringify(single));
+            return Promise.resolve(single);
+        })
+    );
```
```suggestion
    const out = await Promise.all(
        list.map((item) =>
            ctx.cache.tryGet(item.link, async () => {
                const response = await got(item.link, {
                    headers: {
                        Referer: host,
                    },
                });

                const $ = cheerio.load(response.data);
                item.description = $('.wp_articlecontent').html();

                return item;
            })
        )
    );
```
Please use `cache.tryGet` instead of cache.set/get
You can reuse the `item` element return it instead of constructing `single`.

TonyRL reviewed (commented):
Missing `radar.js`
docs: https://docs.rsshub.app/joinus/script-standard.html

dependabot[bot] committed:
chore(deps): bump puppeteer from 17.1.2 to 17.1.3 (#1094)

Bumps [puppeteer](https://github.com/puppeteer/puppeteer) from 17.1.2 to 17.1.3.
- [Release notes](https://github.com/puppeteer/puppeteer/releases)
- [Changelog](https://github.com/puppeteer/puppeteer/blob/main/CHANGELOG.md)
- [Commits](https://github.com/puppeteer/puppeteer/compare/v17.1.2...v17.1.3)

---
updated-dependencies:
- dependency-name: puppeteer
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

hlmu commented:
非常感谢您耐心的检查与建议！❤️我修改下
Tony ***@***.***> wrote on Sep 9, 2022 3:20 AM:

@TonyRL commented on this pull request.

Missing radar.js

________________________________

In lib/v2/hit/hitgs.js<https://github.com/DIYgod/RSSHub/pull/10747#discussion_r966331840>:

> @@ -0,0 +1,64 @@

+const got = require('@/utils/got');

+const cheerio = require('cheerio');

+

+module.exports = async (ctx) => {

+    const host = 'http://hitgs.hit.edu.cn';



⬇️ Suggested change

-    const host = 'http://hitgs.hit.edu.cn';

+    const host = 'https://hitgs.hit.edu.cn';



HTTPS is available

________________________________

In lib/v2/hit/hitgs.js<https://github.com/DIYgod/RSSHub/pull/10747#discussion_r966332215>:

> +    const response = await got({

+        method: 'get',

+        url: host + '/tzgg/list.htm',

+        headers: {

+            Referer: host,

+        },

+    });



got request can be simplified to

⬇️ Suggested change

-    const response = await got({

-        method: 'get',

-        url: host + '/tzgg/list.htm',

-        headers: {

-            Referer: host,

-        },

-    });

+    const response = await got(host + '/tzgg/list.htm', {

+        headers: {

+            Referer: host,

+        },

+    });



________________________________

In lib/v2/hit/hitgs.js<https://github.com/DIYgod/RSSHub/pull/10747#discussion_r966332782>:

> +

+module.exports = async (ctx) => {

+    const host = 'http://hitgs.hit.edu.cn';

+

+    const response = await got({

+        method: 'get',

+        url: host + '/tzgg/list.htm',

+        headers: {

+            Referer: host,

+        },

+    });

+

+    const $ = cheerio.load(response.data);

+    const list = $('.news_list li')

+        .map((i, e) => ({

+            path: $('span a', e).attr('href'),



You can prepare the link property here

⬇️ Suggested change

-            path: $('span a', e).attr('href'),

+            link: host + $('span a', e).attr('href'),



________________________________

In lib/v2/hit/hitgs.js<https://github.com/DIYgod/RSSHub/pull/10747#discussion_r966333559>:

> +module.exports = async (ctx) => {

+    const host = 'http://hitgs.hit.edu.cn';

+

+    const response = await got({

+        method: 'get',

+        url: host + '/tzgg/list.htm',

+        headers: {

+            Referer: host,

+        },

+    });

+

+    const $ = cheerio.load(response.data);

+    const list = $('.news_list li')

+        .map((i, e) => ({

+            path: $('span a', e).attr('href'),

+            pubDate: new Date($('span:nth-child(4)', e).text()).toUTCString(),



Please use parseDate
docs: https://docs.rsshub.app/joinus/pub-date.html

⬇️ Suggested change

-            pubDate: new Date($('span:nth-child(4)', e).text()).toUTCString(),

+            pubDate: parseDate($('span:nth-child(4)', e).text()),



________________________________

In lib/v2/hit/hitgs.js<https://github.com/DIYgod/RSSHub/pull/10747#discussion_r966337762>:

> +    const out = await Promise.all(

+        list.map(async (item) => {

+            const { path, pubDate, title, author } = item;

+            const link = host + path;

+

+            const cache = await ctx.cache.get(link);

+            if (cache) {

+                return Promise.resolve(JSON.parse(cache));

+            }

+

+            const response = await got({

+                method: 'get',

+                url: link,

+                headers: {

+                    Referer: host,

+                },

+            });

+

+            const $ = cheerio.load(response.data);

+            const description = $('.wp_articlecontent').html();

+            const single = {

+                pubDate,

+                author,

+                link,

+                title,

+                description,

+            };

+

+            ctx.cache.set(link, JSON.stringify(single));

+            return Promise.resolve(single);

+        })

+    );



⬇️ Suggested change

-    const out = await Promise.all(

-        list.map(async (item) => {

-            const { path, pubDate, title, author } = item;

-            const link = host + path;

-

-            const cache = await ctx.cache.get(link);

-            if (cache) {

-                return Promise.resolve(JSON.parse(cache));

-            }

-

-            const response = await got({

-                method: 'get',

-                url: link,

-                headers: {

-                    Referer: host,

-                },

-            });

-

-            const $ = cheerio.load(response.data);

-            const description = $('.wp_articlecontent').html();

-            const single = {

-                pubDate,

-                author,

-                link,

-                title,

-                description,

-            };

-

-            ctx.cache.set(link, JSON.stringify(single));

-            return Promise.resolve(single);

-        })

-    );

+    const out = await Promise.all(

+        list.map((item) =>

+            ctx.cache.tryGet(item.link, async () => {

+                const response = await got(item.link, {

+                    headers: {

+                        Referer: host,

+                    },

+                });

+

+                const $ = cheerio.load(response.data);

+                item.description = $('.wp_articlecontent').html();

+

+                return item;

+            })

+        )

+    );



Please use cache.tryGet instead of cache.set/get
You can reuse the item element return it instead of constructing single.

—
Reply to this email directly, view it on GitHub<https://github.com/DIYgod/RSSHub/pull/10747#pullrequestreview-1101278467>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AHF4X5BLTAPKJD4L6FUHOKDV5I4BFANCNFSM6AAAAAAQHUOLGI>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

hlmu committed:
Update lib/v2/hit/hitgs.js

Co-authored-by: Tony <TonyRL@users.noreply.github.com>

hlmu committed:
change code style of hitgs.js

hlmu committed:
add radar.js

hlmu committed:
change http to https for older HIT files

hlmu committed:
Merge branch 'master' of github.com:hlmu/RSSHub

TonyRL committed:
refactor: migrate to v2

TonyRL committed:
docs: fix format

TonyRL committed:
Merge branch 'DIYgod/master' into pr/hlmu/10747

TonyRL committed:
fix: code scan
